junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/TempDirectory.java

```diff
@@ -146,9 +149,10 @@ private void assertSupportedType(String target, Class<?> type) {
 		}
 	}
 
-	private Object getPathOrFile(Class<?> type, ExtensionContext extensionContext) {
+	private Object getPathOrFile(Class<?> type, ExtensionContext extensionContext, String id) {
+		String key = id.isEmpty() ? KEY : KEY + '+' + id;
```

Comment:
⭕ I think we can just use the id and remove `KEY` since we're using our own namespace.

Reply:
Will remove (or inline) the `KEY`.